### PR TITLE
Home work 11

### PR DESCRIPTION
**Pull request by UnitOmega13**

### CODE REVIEW

File: src/Task1.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import java.util.function.IntSupplier;
+import java.util.stream.IntStream;
+
+public class Task1 {
+
+	public static IntStream pseudoRandomStream(int seed) {
```
Comment (Sofasmile) [inline src/Task1.java]: try to simplify this method. we can use "/" and "%" for this

File: src/Task1.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import java.util.function.IntSupplier;
+import java.util.stream.IntStream;
+
+public class Task1 {
```
Comment (Sofasmile) [inline src/Task1.java]: rename to RandomStream or to PseudoStream

File: src/Task1.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+				return mid(current++);
+			}
+		};
+
+		IntStream natural = IntStream.generate(generator);
+
+		return natural;
+	}
+}
+
```
Comment (Sofasmile) [inline src/Task1.java]: remove one blank line after }

File: src/Task2.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import java.util.function.DoubleUnaryOperator;
+
+public class Task2 {
```
Comment (Sofasmile) [inline src/Task2.java]: the same about name

File: src/Task2.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+	public static double integrate(DoubleUnaryOperator f, double a, double b) {
+		int n = 1000000;
```
Comment (Sofasmile) [inline src/Task2.java]: better:
```suggestion
		int n = 1_000_000;
```

File: src/Task2.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+		double dx = (b - a) * 1.0 / n;
```
Comment (Sofasmile) [inline src/Task2.java]: use full name of variables

File: src/Task2.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import java.util.function.DoubleUnaryOperator;
+
+public class Task2 {
+	public static double integrate(DoubleUnaryOperator f, double a, double b) {
```
Comment (Sofasmile) [inline src/Task2.java]: does this method work on stepik?

File: src/Task3.java [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+public class Task3 {
```
Comment (Sofasmile) [inline src/Task3.java]: rename please

Comment (UnitOmega13) [reply src/Task3.java]: Fixed

File: src/FunctionCalculation.java [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import java.util.function.DoubleUnaryOperator;
+
+public class FunctionCalculation {
+	public static double integrate(DoubleUnaryOperator f, double a, double b) {
+		int n = 10_000_000;
+		double differential = (b - a) * 1.0 / n;
+		double integral = 0;
+		double  x;
+		for(int number = 0; number < n ; number++){
+			x = a + number * differential;
```
Comment (Sofasmile) [inline src/FunctionCalculation.java]:
```suggestion
			double x = a + number * differential;
```

File: src/FunctionCalculation.java [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+			integral += f.applyAsDouble(x)*differential;
```
Comment (Sofasmile) [inline src/FunctionCalculation.java]:
```suggestion
			integral += f.applyAsDouble(x) * differential;
```

File: src/RandomStream.java [hunk 11]
```diff
@@ -0,0 +1,40 @@
+import java.util.function.IntSupplier;
+import java.util.stream.IntStream;
+
+public class RandomStream  {
+
+	public static IntStream pseudoRandomStream(int seed) {
```
Comment (Sofasmile) [inline src/RandomStream.java:6]: what about this method?

File: src/FunctionCalculation.java [hunk 12]
```diff
@@ -0,0 +1,15 @@
+import java.util.function.DoubleUnaryOperator;
+
+public class FunctionCalculation {
+	public static double integrate(DoubleUnaryOperator f, double a, double b) {
+		int n = 10_000_000;
+		double differential = (b - a) * 1.0 / n;
```
Comment (Sofasmile) [inline src/FunctionCalculation.java]:
```suggestion
		double differential = (b - a) / n;
```

File: src/RandomStream.java (hunk 11 above)
Comment (UnitOmega13) [reply src/RandomStream.java:6]: He is so named on stepik.

Comment (UnitOmega13) [reply src/RandomStream.java:6]: Also corrected all according to the comments.

Comment (Sofasmile) [reply src/RandomStream.java:6]: try to simplify this method
we can create implementation in one line

Comment (UnitOmega13) [reply src/RandomStream.java:6]: fixed

File: src/FunctionCalculation.java (hunk 12 above)
Comment (Sofasmile) [reply src/FunctionCalculation.java]: what about it?

Comment (UnitOmega13) [reply src/FunctionCalculation.java]: I do not know why, but idea uploads changes through time.

**Review by Sofasmile — APPROVED** [review]

**Pull request closed without merging**